OutOfSchool/OutOfSchool.Common/Constants.cs

```diff
@@ -30,7 +30,11 @@ public static class Constants
 
     public const string EmailRegexViewModel = @"^([\w\.\-]+)@([\w\-]+)((\.(\w){2,3})+)$";
 
-    public const string PasswordRegexViewModel = @"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{6,}$";
+    public const string PasswordRegexViewModel = @"^(?=.*[\p{Lu}])(?=.*[\p{Ll}])(?=.*\d)(?=.*[@$!%*?&])[\p{Lu}\p{Ll}\d@$!%*?&]{8,}$";
```

Comment:
(Usage of regex) Please explain why we have to use such complex regular expression (having few `Zero-width positive lookahead assertion`s [see Regex reference](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference#grouping-constructs) ) instead of implemented check logic directly in the code (as we already have separate validator)? 
I presume that direct logic in code would be both faster and more self-explanatory and let you provide better error message for separate cases ("password is too short", "special symbols not used", etc) .

Reply:
I have only one explanation: I didn’t consider the possibility of replacing the regular expression with checking logic, since the previous code already used a similar regular expression, in addition, after studying Microsoft's recommendations, I added the `GeneratedRegex` attribute for its acceleration. After your recommendation, I see that the direct logic in this case will be more appropriate.
As for the validation error messages, according to the requirements, we should output only one - a general validation error message for all cases, except when password is empty ("Password is required").
Thank you!